### PR TITLE
service.go: parse the /etc/products.d/*.prod files

### PR DESCRIPTION
**Pull request by jvanz**
Adds a code in the service.go to parse the *.prod files. This allow the
user not use the ADDITIONAL_MODULES and parses the file under
/etc/products.d folder. The function uses the file name in  the same
way as  identifier specified in the ADDITIONAL_MODULES environment
variable. Issue #27

Signed-off-by: José Guilherme Vanz <jguilhermevanz@suse.com>

### CODE REVIEW

File: internal/service.go [hunk 1]
```diff
@@ -89,6 +91,21 @@ func moduleEnabledInEnv(identifier string) bool {
...
 
+// moduleEnabledInProductFiles returns true if the provided `identifier` is
+// a name of a file in the  /etc/product.d/*.prod, otherwise false.
+func moduleEnabledInProductFiles(identifier string) bool {
+	files, err := ioutil.ReadDir("/etc/products.d")
+	if err != nil {
+		return false
+	}
+	for _, file := range files {
+		if identifier == strings.TrimSuffix(file.Name(), ".prod") {
```
Comment (saschagrunert) [inline internal/service.go]: Is `.prod` the only extension which can happen there? If so, then I'm fine with that change. :)

If not, then we may remove the extension via:
https://golang.org/pkg/path/filepath/#Ext

File: internal/service.go [hunk 2]
```diff
@@ -89,6 +91,21 @@ func moduleEnabledInEnv(identifier string) bool {
...
 }
 
+// moduleEnabledInProductFiles returns true if the provided `identifier` is
+// a name of a file in the  /etc/product.d/*.prod, otherwise false.
+func moduleEnabledInProductFiles(identifier string) bool {
+	files, err := ioutil.ReadDir("/etc/products.d")
+	if err != nil {
+		return false
+	}
+	for _, file := range files {
```
Comment (saschagrunert) [inline internal/service.go]: Are all files within that directory `.prod` files? Or is there anything else we need to skip inside?

Comment (jvanz) [reply internal/service.go]: As far as I can see in the SUSEConnect and Zypper code, we can find the `baseprodict` which is a link for the `.prod` of the base product.

**Comment by jvanz** [comment]
I've pushed a new  commit considering any file extension and ignoring links in `/etc/products.d`

File: internal/service.go (hunk 2 above)
Comment (saschagrunert) [reply internal/service.go]: Okay, one more thing: Can we only allow `.prod` files there? I have the feeling that we can break things up then some puts something else within that folder...

Comment (jvanz) [reply internal/service.go]: Yeah, my idea in the first commit was consider `.prod` only. But now, I noticed that it did not do that. I'll change the code again to really only consider `.prod` files.

Comment (jvanz) [reply internal/service.go]: Done

**Pull request merged**